tests/server-integration/SetField/SetFieldTest.hs

```diff
@@ -0,0 +1,162 @@
+      NoReqBody
+      (jsonResponse @Value)
+      ( mconcat
+          [ "path" =: ("entry" :: Text)
+          , "field" =: ("public-field" :: Text)
+          ]
+      )
+
+  let fieldMb =
+        responseBody response
+          & scrubDates
+          & findField "public-field"
+
+  case fieldMb of
+    Nothing -> assertFailure $ "Expected field \"public-field\", found nothing"
+    Just value ->
+      value @?=
+        [aesonQQ|
+          {
+            "visibility": "private",
+            "dateModified": "",
+            "contents": "contents"
+          }
+        |]
+
+unit_set_field_public :: IO ()
+unit_set_field_public = cofferTest do
+  unit_set_field_private
+  response <-
+    executeCommand
+      POST
+      ["set-field", "public"]
+      NoReqBody
+      (jsonResponse @Value)
+      ( mconcat
+          [ "path" =: ("entry" :: Text)
+          , "field" =: ("public-field" :: Text)
+          ]
+      )
+
+  let fieldMb =
+        responseBody response
+          & scrubDates
+          & findField "public-field"
+
+  case fieldMb of
+    Nothing -> assertFailure $ "Expected field \"public-field\", found nothing"
+    Just value ->
+      value @?=
+        [aesonQQ|
+          {
+            "visibility": "public",
+            "dateModified": "",
+            "contents": "contents"
+          }
+        |]
+
+unit_set_field_entry_not_found :: IO ()
+unit_set_field_entry_not_found = cofferTest do
+  try @HttpException (setField "entry" "field" "contents") >>=
+    processStatusCodeError status404
+      [aesonQQ|
+        [
+          {
+            "error": "Entry not found at '/entry'.",
+            "code": 300
+          }
+        ]
+      |]
+
+unit_set_field_missing_field_contents :: IO ()
+unit_set_field_missing_field_contents = cofferTest do
+  createEntry "entry"
+
+  try @HttpException
+    (
+      executeCommand
+        POST
+        ["set-field"]
+        (ReqBodyJson Null)
+        (jsonResponse @Value)
+        ( mconcat
+            [ "path" =: ("entry" :: Text)
+            , "field" =: ("field" :: Text)
+            ]
+        )
+    ) >>=
+    processStatusCodeError status400
+      [aesonQQ|
+        [
+          {
+            "error": "The entry at '/entry' does not yet have a field 'field'.\nIn order to create a new field, please include 'FIELDCONTENTS' in the body.",
+            "code": 301
+          }
+        ]
+      |]
+
+unit_set_field_updates_modification_date :: IO ()
+unit_set_field_updates_modification_date = cofferTest do
+  createEntry "entry"
+  setField "entry" "field" "contents"
+
+  t1 <- getCurrentTime
+  response <- setField "entry" "field" "contents2"
+  t2 <- getCurrentTime
+
+  let newModifiedDate = responseBody response ^?! key "dateModified" . _JSON @_ @UTCTime
+  assertBool "Date is not recent"
+    $ t1 <= newModifiedDate && newModifiedDate <= t2
+
+  assertBool "Dates are equal"
+    $ responseBody response ^?! key "dateModified" . _JSON == newModifiedDate
```

Comment:
This check seems redundant.

```hs
let newModifiedDate = responseBody response ^?! key "dateModified" . _JSON @_ @UTCTime
```

```hs
responseBody response ^?! key "dateModified" . _JSON == newModifiedDate
```

Reply:
> `set-field`: updates the entry's and the field's modified date

Hmm, I thought that we want to check all these things.

Reply:
You're already checking those two requirements in lines 157 and 162:

```hs
  assertBool "Date is not recent"
    $ t1 <= newModifiedDate && newModifiedDate <= t2
   ^^^^^^^^^^^^^^^^^ this checks that the entry's date was modified

  assertBool "Dates are not equal"
    $ responseBody response ^?! key "dateModified" . _JSON == newModifiedDate
  assertBool "Dates are not equal"
    $ responseBody response ^?! key "fields" . key "field" . key "dateModified" . _JSON == newModifiedDate
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ this checks that the field's date was modified
```

Reply:
:smile: